ejb3/src/main/java/org/jboss/as/ejb3/component/EJBComponentDescription.java

```diff
@@ -1156,7 +1156,8 @@ public String toString() {
 
         // Next interceptor: run-as-principal
         // Switch users if there's a run-as principal
-        if (runAsPrincipal != null) {
+        // WFLY-11094 @RunAs (runAsRole) sets the principal to anonymous, we should not run this interceptor in that case
+        if (runAsPrincipal != null && !(runAsPrincipal.equals(RunAsPrincipalInterceptor.ANONYMOUS_PRINCIPAL) && runAsRole != null)) {
```

Comment:
This change results in test failures in ```org.jboss.as.test.integration.ejb.security.runas.RunAsTestCase```, ```org.jboss.as.test.integration.ejb.security.RunAsPrincipalTestCase```, and ```org.jboss.as.test.integration.ejb.security.propagation.RunAsWithElytronEJBContextPropagationTestCase```. By default, when using the ```RunAs``` annotation, the anonymous principal needs to be assigned for outgoing calls (as per the spec). So the ```RunAsPrincipalInterceptor``` still needs to be run in this case. If a different principal is needed, the ```RunAsPrincipal``` annotation should be used to specify it.

Reply:
I think this build is using an old commit, all tests pass in local, what I found in the docs is " RunAs Specifies the role which runs the EJB. By default, the EJB runs as the user who actually invokes it." That is why I think if there is a RunAs and not a RunAsPrincipal, the RunASPrincipalInterceptor should not be invoked, I'm investigating why this difference between this build and my local.

Reply:
As mentioned in the [EJB3 Reference Guide](https://docs.jboss.org/author/display/WFLY/EJB+3+Reference+Guide):

"Whenever a run-as role is specified for a given method invocation the default anonymous principal is used as the caller principal. This principal can be overridden by specifying a run-as principal."

Reply:
That sounds good. Thanks @tmiyargi!

Reply:
It turns out the propagation + runas does not work, when RunAsPrincipalInterceptor is executed the AuthenticationConfiguration does not have the credentials, ioFuture is null so it tries an authentication on ConnectionPeerIdentity class ignoring the propagation.

Reply:
It could be a configuration issue in the test (e.g., maybe something like the configured ```remote-outbound-connection``` isn't actually getting used?). Are you able to get this server-to-server scenario working properly manually?

Reply:
I have updated the test and removed the 'fix', it all works fine with the correct configuration and using both annotations together RunAsPrincipal and RunAs.

Reply:
@tmiyargi Looks like the test is still failing on CI.